be/src/exprs/json_functions.cpp

```diff
@@ -1156,6 +1156,204 @@ StatusOr<ColumnPtr> JsonFunctions::_json_keys_without_path(FunctionContext* cont
+    RETURN_IF_COLUMNS_ONLY_NULL(columns);
+    RETURN_IF(columns.size() < 2,
+              Status::InvalidArgument("json_remove requires at least 2 arguments: json_doc and path"));
+
+    size_t rows = columns[0]->size();
+    ColumnBuilder<TYPE_JSON> result(rows);
+    ColumnViewer<TYPE_JSON> json_viewer(columns[0]);
+
+    // Get all path arguments
+    std::vector<ColumnViewer<TYPE_VARCHAR>> path_viewers;
+    for (size_t i = 1; i < columns.size(); i++) {
+        path_viewers.emplace_back(columns[i]);
+    }
+
+    for (size_t row = 0; row < rows; row++) {
+        if (json_viewer.is_null(row)) {
+            result.append_null();
+            continue;
+        }
+
+        JsonValue* json_value = json_viewer.value(row);
+        if (json_value == nullptr) {
+            result.append_null();
+            continue;
+        }
+
+        // Collect all valid paths to remove
+        std::vector<std::string> paths_to_remove;
+        for (size_t path_idx = 0; path_idx < path_viewers.size(); path_idx++) {
+            if (path_viewers[path_idx].is_null(row)) {
+                continue; // Skip null paths
+            }
+
+            std::string path_str = path_viewers[path_idx].value(row).to_string();
+
+            // Parse the JSON path to validate it
+            auto jsonpath = JsonPath::parse(path_str);
+            if (jsonpath.ok()) {
+                paths_to_remove.push_back(path_str);
+            }
+            // Skip invalid paths silently (following MySQL behavior)
+        }
+
+        // Create new JSON with paths removed
+        vpack::Builder builder;
+        ASSIGN_OR_RETURN(auto removed_json, _remove_json_paths(json_value, paths_to_remove, &builder));
+        result.append(std::move(removed_json));
+    }
+
+    return result.build(ColumnHelper::is_all_const(columns));
+}
+
+StatusOr<JsonValue> JsonFunctions::_remove_json_paths(JsonValue* json_value, const std::vector<std::string>& paths,
+                                                      vpack::Builder* builder) {
+    namespace vpack = arangodb::velocypack;
+
+    if (paths.empty()) {
+        // No paths to remove, return original JSON
+        builder->add(json_value->to_vslice());
+        return JsonValue(builder->slice());
+    }
+
+    // Parse all valid paths first
+    std::vector<JsonPath> valid_paths;
+    for (const auto& path_str : paths) {
+        auto jsonpath = JsonPath::parse(path_str);
+        if (jsonpath.ok()) {
+            valid_paths.emplace_back(jsonpath.value());
+        }
+        // Skip invalid paths silently (following MySQL behavior)
+    }
+
+    if (valid_paths.empty()) {
+        // No valid paths to remove, return original JSON
+        builder->add(json_value->to_vslice());
+        return JsonValue(builder->slice());
+    }
+
+    vpack::Slice original_slice = json_value->to_vslice();
+
+    // Build efficient lookup structures
+    std::unordered_set<std::string> exact_paths_to_remove;  // Target level paths
```

Comment:
phmap::flat_hash_set is more efficient?

Reply:
It doesn't matter; the paths provided as parameters to `json_remove()` are typically limited and usually only a few.